kimchi/src/circuits/polynomials/lookup.rs

```diff
@@ -517,6 +520,13 @@ pub fn sorted<
             sorted[i].push(end_val);
         }
 
+        // Duplicate the final sorted value, to fix the off-by-one in the last lookup row.
+        // This is caused by the snakification: all other sorted columns have the value from the
+        // next column added to their end, but the final sorted column has no subsequent column to
+        // pull this value from.
+        let final_sorted_col = &mut sorted[max_lookups_per_row];
+        final_sorted_col.push(final_sorted_col[final_sorted_col.len() - 1].clone());
```

Comment:
I'm struggling to understand what is happening here. 

* we're making the assumption that final_sorted_col is the last column, but that's only true if the tables are taking a single column (and are not divided over several columns)
* we're adding one more value in that last column, but that means the last column will be of size `n + 1 > n` now, and I imagine the last value will be dismissed when interpolated in the domain of size `n`

Reply:
This is because we reverse the columns: this was an edge case before, because the last column was previously not reversed. It actually doesn't matter for this change, but I discovered this in testing with previous PRs, because we push a value to the end of every other column. Off-by-1 errors abound :)